other/upgrade.php

```diff
@@ -5761,4 +5761,63 @@ function doTheDelete(theCheck)
 			Simple Machines';
 }
 
+/**
+ * Convert MySQL (var)char ip col to binary
+ * newCol needs to be a varbinary(16) null able field
+ * return true or false
+ */
+function MySQLConvertOldIp($targetTable,$oldCol,$newCol)
+{
+	//mysql default max length is 1mb http://dev.mysql.com/doc/refman/5.1/en/packet-too-large.html
+	$max = 10000;
+	$arIp = array();
+	
+	$request = $smcFunc['db_query']('', 'DROP TABLE IF EXISTS {db_prefix}ip_table');
+	$smcFunc['db_free_result']($request);
+	
+	$request = $smcFunc['db_query']('', 'CREATE TABLE {db_prefix}ip_table
+    (oldip varchar(255), newip varbinary(16) )
+    ENGINE = MEMORY');
+	$smcFunc['db_free_result']($request);
+	
+	$request = $smcFunc['db_query']('', 'select distinct '.$oldCol.' from {db_prefix}'.$targetTable);
+	while($row = $smcFunc['db_fetch_assoc']($request))
+		$arIp[] = $row[$oldCol]; 
+	$smcFunc['db_free_result']($request);
+	
+	$insertStart = 'insert into {db_prefix}ip_table(oldip, newip) values';
+	$query = $insertStart;
+	$impArray = array();
+	$x = 0;
+	for($i = 0; $i < length($arIp); $i++)
```

Comment:
( ! ) Fatal error: Call to undefined function length() in D:\www\SMF2.1\upgrade.php on line 5794